deployments/cluster/educator.nix

```diff
@@ -11,45 +11,70 @@ in
   ##
   # `map` to make additional SGs easier to add and SG list more readable
   deployment.ec2.securityGroupIds = map (x: resources.ec2SecurityGroups."cluster-${x}-sg".name ) (
-    [ "educator-api-private" "witness-api-private" ]
+    [ "educator-api-private" "witness-public" ]
   );
 
   networking.firewall.allowedTCPPorts = [
-    4040 4041   # Witness ZMQ API
+    4010 4011   # Witness ZMQ API
     4030        # Witness HTTP Wallet API
     4040        # Educator HTTP API
   ];
 
-  services.disciplina = rec {
+  services.disciplina = let
+    config-key = "alpha";
+
+  in rec {
     enable = true;
     type = "educator";
 
+
+    config."${config-key}" = rec {
+      witness = rec {
+        appDir = "/var/lib/disciplina-${type}";
+        db.path = "${appDir}/witness.db";
+        api.addr = "0.0.0.0:4030";
+        keys = {
+          type = "basic";
+          path = "${appDir}/witness.key";
+          genNew = true;
+        };
+      };
+
+      educator = {
+        publishing.period = "30s";
+        db.mode = {
+          path = "${witness.appDir}/educator.sqlite";
+          connNum = 4;
+          maxPending = 100;
+        };
+
+        keys = {
+          path = "${witness.appDir}/educator.key";
+          genNew = true;
+        };
+
+        api = {
+          serverParams.addr = "0.0.0.0:4040";
+          botParams = {
+            operationsDelay = "3s";
+            seed = "super secure"; # I don't know whether or not this is sensitive data
+          };
+          studentAPINoAuth.enabled = false;
```

Comment:
does that have auth or not?

Reply:
This is one of those options that should just have a sane default option. Enabling this option disables auth. Confusing.